app/services/github_service.rb

```diff
@@ -0,0 +1,48 @@
+class GithubService < BaseService
+  def initialize(payload:, event:)
+    @payload = payload
+    @event = event
+  end
+
+  def call
+    handle_request
+  end
+
+  private
+
+  def handle_request
+    build_project
+    handle_event
+    handle_action
+  end
+
+  def build_project
+    repo = @payload['repository']
+    @project = Project.find_or_create_by!(github_id: repo['id']) do |pj|
+      pj.name = repo['name']
+      pj.description = repo['description']
+    end
+  end
+
+  def handle_event
+    event = Event.create!(project: @project, data: @payload, name: @event)
+    raise StandardError unless handleable_event?
+
+    @event_type = find_or_create_event_type
+    event.update!(handleable: find_or_create_event_type)
```

Comment:
Can't you use `@event_type` here instead of calling `find_or_create_event_type` again?

Reply:
Yes, my bad!